tree/master/cloud/src/solution/Deployment/Scripts/NewMetadata.ps1

```diff
@@ -0,0 +1,353 @@
+	# Write into log file
+	if(-not [String]::IsNullOrEmpty($Message)) {
+		($Message) | Out-File $LogFile -Append
+	}
+}
+
+Function Get-ContentTypeId
+{
+    foreach( $cc in $ctx.Web.ContentTypes)
+    {
+        if($cc.Name -eq $newContentTypeName)
+        {
+            Write-Host "New Content Type Name exists. Adding site columns to this content type"
+            $global:isNewContentTypeExists = $true
+            $global:contentTypeId = $cc.Id.StringValue
+            break;
+        }   
+    }
+}
+
+#function for creating a new content type in content type hub site collection
+function New-SPOContentType
+{
+    param(        
+        [Parameter(Mandatory=$true,Position=1)] [string]$Description,
+        [Parameter(Mandatory=$true,Position=2)] [string]$Name,
+        [Parameter(Mandatory=$true,Position=3)] [string]$Group,
+        [Parameter(Mandatory=$true,Position=4)] [string]$ParentContentTypeId
+    ) 
+    $lci =New-Object Microsoft.SharePoint.Client.ContentTypeCreationInformation
+    $lci.Description=$Description
+    $lci.Name = $Name
+    $lci.ParentContentType=$ctx.Web.ContentTypes.GetById($ParentContentTypeId)
+    $lci.Group=$Group
+    $ContentType = $ctx.Web.ContentTypes.Add($lci)
+    $ctx.Load($contentType)    
+    try
+    {        
+       $ctx.ExecuteQuery()      
+       Write-Host "Content Type " $Name " has been added to site collection  " $contentTypeHubUrl
+    }
+    catch [Net.WebException]
+    {
+       Write-Host $_.Exception.ToString()
+    }
+} 
+
+#Function to create site columns and add those site columns to the corresponding content type
+function Add-SPOSiteColumns
+{
+    param(
+    [Parameter(Mandatory=$true,Position=1)] [string]$columnName,
+    [Parameter(Mandatory=$true,Position=2)] [string]$columnType,
+    [Parameter(Mandatory=$true,Position=3)] [string]$groupName,
+    [Parameter(Position=4)] [string]$subColumnType,
+    [Parameter(Position=5)] [string]$values,
+    [Parameter(Position=6)] [string]$contentTypeId
+    )
+    $spWeb = $ctx.Web;
+    $fields = $spWeb.Fields;
+    $ctx.load($fields)
+    try{
+        $ctx.executeQuery()            
+    }
+    catch{            
+    }    
+    foreach($field in $fields){      
+        if ($field.internalname -eq $columnName){            
+            Show-Message -Message "Column   $columnName already exists in the site collection. "  -Type ([MessageType]::Warning)  -Newline $true
+            $columnExists = 1  
+            break
+        }
+        else{
+            $columnExists = 0            
+        }       
+    }
+
+    #If the column does not exists, create that site column
+    if ($columnExists -eq 0){
+        Show-Message -Message "Column  $columnName does not exists in the site collection. Adding to the site collection"  -Type ([MessageType]::Info)  -Newline $true
+        $fieldAsXML = ''
+        #For Text, DateTime and Boolean
+        if( $columnType -eq "Text" -or $columnType -eq "DateTime" -or $columnType -eq "Boolean"){
```

Comment:
@premchandpl  use existing libraries instead of adding this new script. or else remove all utility libraries and use powershell script only for entire deployment.